orders/src/main/resources/static/js/orders.js

```diff
@@ -0,0 +1,151 @@
+window.productsByName = new Map();
+window.productsByBarcode = new Map();
+window.productsByCategory = new Map();
+window.finalOrders = new Set();
+
+function removeOptions(selectbox) {
+    var i;
+    for (i = selectbox.options.length - 1; i >= 0; i--) {
+        selectbox.remove(i);
+    }
+}
+
+//  initPage
```

Comment:
put the init part in a function, and add an "onload" event to the HTML

Reply:
That was the case before I put the JS on another file. Once the JS file is external to the HTML, for some reason it cannot use thymeleaf's [[${attributeName}]] reference for some reason - the webpage throws a syntax error for unexpected {.
The only workaround I found was to add a global js variable (under the window object) in script on the HTML page, and that happened at the body - so I deleted the onload. I'll move the script to the head, that might work.

Reply:
Did you try sending it as a function parameter? (onload=myJsFunction(${orders}) )

Reply:
yes, didn't work. welcome to try - i'll be glad if you find howand tell, as i've looked through the internet in search for a way to use spring's MVC in an external js file, and for the life of me didn't find something that worked.

Reply:
I looked at your code - you moved the init function away to the JS page, that's great. 
Also, the internet claims you are correct and there's no other way to do this